### ISSUE TITLE
Use Scratch.jl for version updates

### ISSUE THREAD

**Issue by Libbum** [JuliaDynamics/Agents.jl#493]
> From a discussion on Slack that will be forever lost, it is better to use https://github.com/JuliaPackaging/Scratch.jl to display update messages instead of the current pipeline with `touch`. Stefan Karpinski said that
> 
> > We really don’t want people modifying package code in place cry
> 
> and also cited other related problems, while Elliot Saba further commended that the approach will be impossible in the future:
> 
> > We’re eventually going to make packages read-only.
> 
> Elliot thankfully pointed out that the replacement is easy:
> 
> > https://github.com/JuliaPackaging/Scratch.jl , Basically, instead of using the DIR macro, you would use the get_scratch macro, that's the only change. But you get a lot of benefits, including that it will work even if your package is included in PackageCompiler output. The scratch directory can be cleared, if for example the package is uninstalled then reinstalled later. The user can also manually clear it.

&mdash;JuliaDynamics/DrWatson.jl#246

**Comment by fbanning** [comment]
I don't think that I understand the benefits of using Scratch.jl, could you maybe ELI5 it for me?

**Comment by Libbum** [comment]
```shell
/repos/Agents.jl/src % ls -l
total 32K
-rw-r--r-- 1 tim users 2.5K Jun  6 19:52 Agents.jl
drwxr-xr-x 2 tim users 4.0K Apr 30 14:19 core/
-rw-r--r-- 1 tim users 1.1K Apr 30 14:19 deprecated.jl
drwxr-xr-x 2 tim users 4.0K Jun  5 18:28 models/
drwxr-xr-x 2 tim users 4.0K Jun  5 18:23 simulations/
drwxr-xr-x 2 tim users 4.0K Apr 30 14:19 spaces/
drwxr-xr-x 4 tim users 4.0K Jun  5 18:28 submodules/
-rw-r--r-- 1 tim users    0 Jan 25 21:07 update_v4.0
-rw-r--r-- 1 tim users    0 Feb 22 21:43 update_v4.1
-rw-r--r-- 1 tim users    0 Mar 17 09:46 update_v4.2
drwxr-xr-x 2 tim users 4.0K Apr 30 14:19 visualization/
```

See all those update files? We track their creation and show an update message in the repl when we need to showcase changes and features. The files just sit around and pollute the folder. 

It's a bit of a hack, and anyhow: there are some changes happening upstream that'll make this implementation not even possible. Exactly what scratch does and provides I'm not sure - haven't looked just yet, but it's meant to be a drop-in replacement for these update files AFAIK.
